Code Review

SFE-Brudnerd opened the pull request:
Please review the code in the Codeless BLE folder under Firmware. 

Optionally, please review the tests if you want to.

SFE-Brudnerd committed:
Add tests. All but QSPI and BLE working.

SFE-Brudnerd committed:
Use Serial2.

SFE-Brudnerd committed:
Add Codeless BLE basic wrapper.
Add examples, test.
Add arduino library files.
Add gitignore.
Add license.

SFE-Brudnerd committed:
Update for Reds.

Change QSPI flash chip to same one as Portenta C33
Remove boot button
Move reset button to space it out
Add boot test point
Recenter programming header silk
Update GPIO silk
Rename nets in schematic
Remove BLE Reset jumper
Remove DNP caps from 24MHz xtal

SFE-Brudnerd committed:
Add 6-board panel files.

gigapod reviewed (approved):
I just performed a cusory review - there is a lot here - and it looks solid/clean.

SFE-Brudnerd committed:
Add CodelessBLE zip for arduino.

SFE-Brudnerd committed:
Add QSPI Datasheet

SFE-Brudnerd committed:
fix parsePipeData bug

SFE-Brudnerd committed:
V10 hardware post reds review update

Connect fills on bottom side near bottom diamond.
Add user button near reset button.
VREF EN now pulls up to 3.3V instead of VIN.
Change VREF silk bottom side to inverted VA.
Update EN and !RST! silk flags to use the right arrow.
Move vias to allow fills.
Adjust guard trace around Y1 to accommodate removing caps.
Add ground vias near QSPI data lines where possible.

SFE-Brudnerd committed:
Rename silk

VA to VR.

SFE-Brudnerd committed:
Remove unused default folders.

SFE-Brudnerd committed:
Repanelize post reds review.

sfe-SparkFro reviewed (commented):
Overall seems fine. Could do with a lot more comments everywhere. Also not really a fan of commented out code, IMO code should be deleted if it's not used. Also saw a mix of case conventions; a lot of variables use snake case, but some are camel case (I believe camel case is the convention to use here).

SFE-Brudnerd committed:
Update Codeless Firmware.
Update library.properties with correct URL.
Add comments and header to examples.
Remove unused code in examples.
Remove magic numbers in examples and libraries.
Comment code.
Refactor debug statements to improve readability.
Refactor logic to reduce nesting.

SFE-Brudnerd committed:
Update examples.
Refactor examples to use new function names.
Rezip library post changes.

SFE-Brudnerd committed:
Remove WIP binary mode code.

SFE-Brudnerd committed:
Update SD test to remove SD power control.
